File: majorProject/src/main/java/com/aocc/framework/implementation/TouchHandler.java
package com.aocc.framework.implementation;

import java.util.List;

import android.view.View.OnTouchListener;

import com.aocc.framework.Input.TouchEvent;

import com.aocc.framework.Viewport;

public interface TouchHandler extends OnTouchListener {
    void setViewport(Viewport viewport);

    public boolean isTouchDown(int pointer);
    
    public int getTouchX(int pointer);
    
    public int getTouchY(int pointer);
    
    public List<TouchEvent> getTouchEvents();
}

File: majorProject/src/main/kotlin/com/aocc/framework/implementation/TouchHandler.kt
package com.aocc.framework.implementation

import android.view.View
import com.aocc.framework.Input
import com.aocc.framework.Viewport

interface TouchHandler : View.OnTouchListener {
    fun setViewport(viewport: Viewport)

    fun isTouchDown(pointer: Int): Boolean

    fun getTouchX(pointer: Int): Int

    fun getTouchY(pointer: Int): Int

    val touchEvents: List<Input.TouchEvent>
}
